src/views/EditSidebar.vue

```diff
@@ -107,7 +107,7 @@
 				@update-end-timezone="updateEndTimezone"
 				@toggle-all-day="toggleAllDay" />
 
-			<InvitationResponseButtons v-if="isViewedByAttendee && userAsAttendee && !isReadOnly"
+			<InvitationResponseButtons v-if="!isReadOnly && isViewedByAttendee && userAsAttendee"
```

Comment:
**issue:** This is a valid work around, but I prefer if you fixed instead `userAsAttendee` to handle `getCurrentUserPrincipalEmail` being undefined (so that it returns `null` straight away). Then there can be other uses of this computed property in other places without adding another check.
https://github.com/nextcloud/calendar/blob/a23e32dadda4a7751f6cc057a194df317cf19d56/src/mixins/EditorMixin.js#L229-L241

Also, the `userAsAttendee` shouldn't be needed, as `isViewedByAttendee` does basically that.

Reply:
I see, we can simply move the read-only check to `userAsAttendee`.
Cleaned up the checks, hope it does not cause any other issues

Reply:
Please still add handling for `this.$store.getters.getCurrentUserPrincipalEmail` being null, to avoid unexpected cases

Reply:
getCurrentUserPrincipalEmail will fail if currentUserPrincipal is null so a simple check for `this.$store.state.principals.currentUserPrincipal === null` would be fine

I added a more general solution via an additional check in https://github.com/nextcloud/calendar/blob/8df701985580c213da0eb4b37abc3debae77381d/src/store/principals.js#L103